### PR TITLE
Adding a ttlUpdated field in MessageInfo

### PR DESCRIPTION
**Pull request by vgkholla**
Should be `false` for all existing use cases.
This is the read change - the write change will be made as a different commit to allow for smooth transitions.

### CODE REVIEW

File: ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java [hunk 1]
```diff
@@ -130,6 +136,13 @@ public boolean isDeleted() {
     return isDeleted;
   }
 
+  /**
+   * @return @code true} if the message's ttl has been updated, {@code false} otherwise
```
Comment (dharju) [inline ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: Missing '{' in front of '@code'

Comment (vgkholla) [reply ambry-api/src/main/java/com.github.ambry/store/MessageInfo.java]: Fixed

**Comment by vgkholla** [comment]
@dharju comment addressed

**Comment by vgkholla** [comment]
@dharju reminder to review

**Comment by vgkholla** [comment]
@dharju you are only reviewer - so if you are ok with it, please merge

**Pull request merged**